passes/techmap/clockgate.cc

```diff
@@ -37,6 +39,161 @@ ClockGateCell icg_from_arg(std::string& name, std::string& str) {
 	return c;
 }
 
+static std::pair<std::optional<ClockGateCell>, std::optional<ClockGateCell>>
+	find_icgs(std::string filename) {
+	std::ifstream f;
+	f.open(filename.c_str());
+	if (f.fail())
+		log_cmd_error("Can't open liberty file `%s': %s\n", filename.c_str(), strerror(errno));
+	LibertyParser libparser(f);
+	f.close();
+	auto ast = libparser.ast;
+
+	// We will pick the most suitable ICG absed on tie_lo count and area
+	struct ICGRankable {
+		ClockGateCell icg;
+		double area;
```

Comment:
Let's move area into `ClockGateCell` even if it will be unfilled for non-liberty invocations of the command

Reply:
I don't like this idea, ClockGateCell represents a cell's interface. The more types the merrier

Reply:
I disagree, it makes the code less readable due to superfluous `.icg`, but I won't withhold approving the PR over it

Reply:
solved with inheritance instead